internal/proxy/providers/sso.go

```diff
@@ -105,10 +106,11 @@ func (p *SSOProvider) Redeem(redirectURL, code string) (*SessionState, error) {
 	params.Add("code", code)
 	params.Add("grant_type", "authorization_code")
 
-	req, err := newRequest("POST", p.RedeemURL.String(), bytes.NewBufferString(params.Encode()))
+	req, err := newRequest("POST", p.ProxyRedeemURL.String(), bytes.NewBufferString(params.Encode()))
 	if err != nil {
 		return nil, err
 	}
+	req.Host = p.RedeemURL.Host
```

Comment:
Trying to understand the necessity for the ProxyRedeemURL - why are we setting the request host to the RedeemURL's host here?

Reply:
the sso-auth is making decisions on serving the content based on the host.  doing this let's the sso-auth know that the traffic is for that http server.  i didn't dig into it on the sso-auth end and figured it would be easier to set this host header here then to require the sso-auth to also have the `PROXY_PROVIDER_URL` var fed to it so it could know addtional virtual server addresses it should be responding to.  

without this host header set the sso-auth throws a 4xx error and the sso-proxy redeem fails.

Reply:
I think that's here on the sso-auth end? https://github.com/buzzfeed/sso/blob/master/internal/auth/authenticator.go#L603